### ISSUE TITLE
[8.0] TypeError: _setup_fields() got an unexpected keyword argument 'partial'

### ISSUE THREAD

**Issue by ghost** [OCA/geospatial#38]
When I start the installation of the module : Geospatial support for OpenERP

I got this message : "TypeError: _setup_fields() got an unexpected keyword argument 'partial'"

Can anyone help ? 

Full thanks,

**Comment by krachtvoll** [comment]
Got the same problem here.

This is the full error:

Traceback (most recent call last):
  File "/home/openerp/odoo/openerp/http.py", line 500, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/openerp/odoo/openerp/http.py", line 517, in dispatch
    result = self._call_function(*_self.params)
  File "/home/openerp/odoo/openerp/http.py", line 283, in _call_function
    return checked_call(self.db, *args, *_kwargs)
  File "/home/openerp/odoo/openerp/service/model.py", line 113, in wrapper
    return f(dbname, _args, *_kwargs)
  File "/home/openerp/odoo/openerp/http.py", line 280, in checked_call
    return self.endpoint(_a, *_kw)
  File "/home/openerp/odoo/openerp/http.py", line 733, in __call__
    return self.method(_args, *_kw)
  File "/home/openerp/odoo/openerp/http.py", line 376, in response_wrap
    response = f(_args, *_kw)
  File "/home/openerp/odoo/addons/web/controllers/main.py", line 948, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/openerp/odoo/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, _args, *_kwargs)
  File "/home/openerp/odoo/openerp/api.py", line 237, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/home/openerp/odoo/openerp/addons/base/module/module.py", line 450, in button_immediate_install
    return self._button_immediate_function(cr, uid, ids, self.button_install, context=context)
  File "/home/openerp/odoo/openerp/api.py", line 237, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/home/openerp/odoo/openerp/addons/base/module/module.py", line 498, in _button_immediate_function
    registry = openerp.modules.registry.RegistryManager.new(cr.dbname, update_module=True)
  File "/home/openerp/odoo/openerp/modules/registry.py", line 324, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/openerp/odoo/openerp/modules/loading.py", line 358, in load_modules
    loaded_modules, update_module)
  File "/home/openerp/odoo/openerp/modules/loading.py", line 263, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/home/openerp/odoo/openerp/modules/loading.py", line 162, in load_module_graph
    registry.setup_models(cr, partial=True)
  File "/home/openerp/odoo/openerp/modules/registry.py", line 167, in setup_models
    model._setup_fields(cr, SUPERUSER_ID, partial=partial)
  File "/home/openerp/odoo/openerp/api.py", line 237, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/home/openerp/odoo/openerp/api.py", line 332, in old_api
    result = method(recs, _args, *_kwargs)
TypeError: _setup_fields() got an unexpected keyword argument 'partial'

Running Ubuntu 14.04.2 LTS
psql (PostgreSQL) 9.3.8
Odoo 8.0

**Comment by krawser** [comment]
Hi

I also get the error:
TypeError: _setup_fields() got an unexpected keyword argument 'partial'

When trying to install one of the geoengine / geospatial apps....

Thanks

**Comment by nbessi** [comment]
Hello, 

It is probably due to a change in Odoo "core". It need some time to investigate.

**Comment by ghost** [comment]
Is there any progress on this issue, we have encountered the same issue

**Comment by lmignon** [comment]
@befksbuild no progress since I'm not able to reproduce the problem on a fresh install.

**Comment by muralikrishnareddy** [comment]
Same issue for me too 
_setup_fields() got an unexpected keyword argument 'partial'

**Comment by yvaucher** [comment]
@infocrea 
@muralikrishnareddy 
@befksbuild 

TL; DR;
You need to update your openerp 8.0 to at least revision https://github.com/OCA/OCB/commit/5fee95ca6366954839e6db0cec5cf1b9b06c023d
or
https://github.com/odoo/odoo/commit/5fee95ca6366954839e6db0cec5cf1b9b06c023d

But you certainly should update to the last version of odoo as you might have more than 13 month of missing fixes.

## 

After some investigation, on the error raised by this code:
https://github.com/OCA/geospatial/blob/8.0/base_geoengine/geo_view/ir_view.py#L32-L41

I found that it matches with the last revision code in openerp:
https://github.com/OCA/OCB/blob/e62b9e7c93236ecfbabe4c03d49ddd622c2b7f71/openerp/modules/registry.py#L199

A change occured here on 14 Jan 2015 which changed the signature of the `_setup_fields` function removing `partial` arguments from it.
https://github.com/OCA/OCB/commit/5fee95ca6366954839e6db0cec5cf1b9b06c023d#diff-c1b8488f841717177876eb16877596af

Thus if you server is not up to date, the odoo code will call the geoengine override with an extra argument. And thus will fail.

Thus I close this issue.
